subsys/event_manager/event_manager.c

```diff
@@ -115,112 +106,18 @@ static void log_event_init(void)
 	}
 }
 
-static void trace_event_execution(const struct event_header *eh, bool is_start)
-{
-	size_t event_cnt = __stop_event_types - __start_event_types;
-	size_t event_idx = event_cnt + (is_start ? 0 : 1);
-	size_t trace_evt_id = profiler_event_ids[event_idx];
-
-	if (!IS_ENABLED(CONFIG_EVENT_MANAGER_TRACE_EVENT_EXECUTION) ||
-	    !is_profiling_enabled(trace_evt_id)) {
-		return;
-	}
-
-	struct log_event_buf buf;
-	ARG_UNUSED(buf);
-
-	profiler_log_start(&buf);
-	profiler_log_add_mem_address(&buf, eh);
-	profiler_log_send(&buf, trace_evt_id);
-}
-
-static void trace_event_submission(const struct event_header *eh)
-{
-	if (!IS_ENABLED(CONFIG_EVENT_MANAGER_PROFILER_ENABLED)) {
-		return;
-	}
-
-	const struct event_type *et = eh->type_id;
-	size_t event_idx = et - __start_event_types;
-	size_t trace_evt_id = profiler_event_ids[event_idx];
-
-	if (!et->ev_info ||
-	    !et->ev_info->profile_fn ||
-	    !is_profiling_enabled(trace_evt_id)) {
-		return;
-	}
-
-	struct log_event_buf buf;
-	ARG_UNUSED(buf);
-
-	profiler_log_start(&buf);
-
-	if (IS_ENABLED(CONFIG_EVENT_MANAGER_TRACE_EVENT_EXECUTION)) {
-		profiler_log_add_mem_address(&buf, eh);
-	}
-	if (IS_ENABLED(CONFIG_EVENT_MANAGER_PROFILE_EVENT_DATA)) {
-		et->ev_info->profile_fn(&buf, eh);
-	}
-
-	profiler_log_send(&buf, trace_evt_id);
-}
-
-static void trace_register_execution_tracking_events(void)
+void __weak event_manager_trace_event_execution(const struct event_header *eh,
```

Comment:
Please add that to header so there would be a basic check when function is overloaded